src/erl/typespec.rs

```diff
@@ -0,0 +1,263 @@
+use crate::{ast::*, erl::format::*, erl::literals::atom, erl::names, pretty::*, typ::Type};
+
+use heck::{CamelCase, SnakeCase};
+use itertools::Itertools;
+
+// Creates the contents of the `export_type([..])` line
+pub fn type_exports(module: &TypedModule) -> Document {
+    concat(
+        module
+            .statements
+            .iter()
+            .flat_map(|s| match s {
+                Statement::CustomType {
+                    public: true,
+                    name,
+                    constructors,
+                    ..
+                } => {
+                    // We only count the params that we can find in the constructors as we can only
+                    // put those ones into the type definition as otherwise we'll have unused type
+                    // param warnings
+                    let constructor_params = extract_params_from_constructors(&constructors);
+                    Some((name.to_snake_case(), constructor_params.len()))
+                }
+
+                _ => None,
+            })
+            .map(|(n, a)| atom(n).append("/").append(a))
+            .intersperse(", ".to_doc()),
+    )
+}
+
+pub fn type_name_to_atom(name: &str) -> String {
+    let name = &name.to_snake_case();
+    if names::is_erlang_reserved_word(name) {
+        format!("'{}'", name)
+    } else {
+        name.to_string()
+    }
+}
+
+/* Generate an erlang '-type' line for this custom type. There are three options for entries:
```

Comment:
Nice docs! Rust doc comments start with `///`, so let's add that so it shows up in rendered docs and IDEs

Reply:
Cool!